### ISSUE TITLE
eBooks of menu scrolltexts

### ISSUE THREAD

**Issue by nguillaumin** [atari-legend/atari-legend#143]
Hi @stgraveyard @Brume-AL @MugUK,

I've been exploring the idea of generating eBooks for the menu sets, to be able to view and read all scrolltexts at once on a more pleasant device than a computer screen. These would be downloadable for each menuset.

Please see the prototype below and let me know what you think? Good idea or not?

![IMG_20210513_151936049](https://user-images.githubusercontent.com/762512/118131969-74a5fe00-b3ff-11eb-837a-7c0c3d7db072.jpg)
![IMG_20210513_151941307](https://user-images.githubusercontent.com/762512/118131971-75d72b00-b3ff-11eb-9fc2-73bb2d9efed3.jpg)
![IMG_20210513_151950803](https://user-images.githubusercontent.com/762512/118131973-75d72b00-b3ff-11eb-8413-ed04b74cd72b.jpg)
![IMG_20210513_151958812](https://user-images.githubusercontent.com/762512/118131975-766fc180-b3ff-11eb-9bd8-18fb4ba01975.jpg)

**Comment by nguillaumin** [comment]
Attaching an example one: [Scrolltexts of Adrenalin UK.zip](https://github.com/atari-legend/atari-legend/files/6472700/Scrolltexts.of.Adrenalin.UK.zip)

**Comment by MugUK** [comment]
Yes but the technical author that beats inside me would want to reformat them all to be formatted in sentence case to enable legibility :)

We would definitely need to redact all addresses.

**Comment by stgraveyard** [comment]
My god, you are amazing.

Tell me, how does this work? I mean, if I would anounce this on facebook and twitter with those screenshots, I think people will go wild :D 

Have you written code to generate these ebook files automatically.

**Comment by stgraveyard** [comment]
> Yes but the technical author that beats inside me would want to reformat them all to be shown in sentence case to enable legibility :)
> 
> We would definitely need to redact all addresses.

That would be overkill I guess :D

**Comment by nguillaumin** [comment]
> reformat them all to be shown in sentence case to enable legibility

Yes the casing is a bit annoying, but it is respecting the original data, so to me it's fine, if we think of it more as a "preservation" thing. I did come across some scrolltexts that have "proper" casing

> redact all addresses.

Not sure, they're already published on the site. Most of them are probably irrelevant/obsolete? I guess we could have a notice to tell people they can contact us if they want the address removed but I would not necessarily redact them as a rule?

> Tell me, how does this work?

It does generate them on the fly, using a PHP library to generate ePub file. Then you download and you copy to your device for reading. There are ePub reader for Android (and IOS I assume) and you can also copy to actual ebook reader like in my pictures.

I have put it on the development site so that you can play with it:

![1](https://user-images.githubusercontent.com/762512/118248550-b341c400-b4a4-11eb-8b16-86ba81f1d8ee.png)

**Comment by nguillaumin** [comment]
Hm, just saw it does not work on the development site, I'll have a look.

**Comment by MugUK** [comment]
I'd best start ripping some more then :)

**Comment by nguillaumin** [comment]
Ok should work now.

**Comment by Brume-AL** [comment]
I'm not a ebook user, but it looks very good!
Congratulation Nico, another brilliant job 👍👍

**Comment by stgraveyard** [comment]
I agree with Bruno. I don't know a thing about E readers and E books, but this is just so cool as an extra. 
I'm so happy you are on the team :-) 
Please continue to amaze me, I'm gonna let the people already know you are working on this, it is just so cool!

**Comment by nguillaumin** [comment]
Thanks :wink: Even if you're not into eBooks, I think it's a convenient way to read a bunch of scrolltexts on your phone via an eBook reader app on a phone, in the train for example, rather than via the website. Anyway, it's just a little extra for those who want to use it.

One thing I need help with: I can set a "cover" image for the eBooks which would be nicer than just the title page I have for now. However I suck at Photoshop! It would be great if someone could draw a cover image using the AL logo / colors. The aspect ratio of the image needs to be book-like (like 1.6:1). There would be the need for some space in the middle where I could overlay some text like the name of the menu set and the crews.

**Comment by Brume-AL** [comment]
A very short message to indicate that the name of the site is 'Atari Legend' (there is a space between both words).

**Comment by nguillaumin** [comment]
Thanks, that was never completely clear to me. I will fix all the places where it's mentioned without a space.

**Comment by nguillaumin** [comment]
Here's what I came up with for the cover so far:

![1](https://user-images.githubusercontent.com/762512/118357622-5e24b180-b57b-11eb-8cfc-65f0523df51c.png)

**Comment by stgraveyard** [comment]
Where is this graphic placed exactly Nico? I'm not following here at all I'm afraid. Wasn"t it a zip file you download to use on phone or ebook reader?

**Comment by MugUK** [comment]
Are we also going to include a link to the site as well as a link to the menu page where the scrolltexts came from?

**Comment by nguillaumin** [comment]
The graphic is the "cover" of the book. It is the first page if you will, and also what is displayed on an e-book reader when it's in sleep mode, something like this:

![2](https://user-images.githubusercontent.com/762512/118365877-c3d66500-b59e-11eb-8579-8ddf14cd4174.jpg)

It's also what's displayed in the "virtual library", either on the e-book reader or on an app on Android, for example:

![3](https://user-images.githubusercontent.com/762512/118365966-131c9580-b59f-11eb-9a7f-1013cabab2a0.jpg)


It's not strictly needed, but it's nice to have.

> include a link to the site

There is a link to AL (see the very first pic). I can also put a link to the menuset.

**Comment by MugUK** [comment]
I've installed SumatraPDF reader for Windows on my PC. The following points may not occur on other PDF readers / different menu group exports :)

1) The centralised text on the Cover Page

![2021-05-15 (2)](https://user-images.githubusercontent.com/17252699/118374891-3141c000-b5b6-11eb-9dd2-1a523fd07f46.png)

2) The large font for the scrolltext - can we make it a point or two smaller?

![2021-05-15 (4)](https://user-images.githubusercontent.com/17252699/118374908-43bbf980-b5b6-11eb-9d10-4da49d8fd6de.png)


3) When there's no scrolltext, can we include a carriage return between the menu contents and the "No scrolltext found" - and don't central format the message either :)

![2021-05-15 (3)](https://user-images.githubusercontent.com/17252699/118374919-4dddf800-b5b6-11eb-8018-493f5a26f1e1.png)

**Comment by MugUK** [comment]
4) If there's too much text, it's chopped off :(

![2021-05-15 (5)](https://user-images.githubusercontent.com/17252699/118375023-f0967680-b5b6-11eb-84a5-05cf529e2eed.png)

**Comment by MugUK** [comment]
5) If a menu is missing can it be made on a separate line (like "No scrolltext found") but make it appear bold (or in red, so long as it appears bold) even on monochrome e-readers?

![2021-05-15 (7)](https://user-images.githubusercontent.com/17252699/118375089-6b5f9180-b5b7-11eb-9b16-db8a25b7af8a.png)

**Comment by MugUK** [comment]
6) Final query - what happened to page 137? You can only go to page 136 in this Awesome/Elite export.

![2021-05-15 (8)](https://user-images.githubusercontent.com/17252699/118375194-022c4e00-b5b8-11eb-8132-8fa4a823ab5c.png)

**Comment by nguillaumin** [comment]
Thanks for the thorough testing. Unfortunately most of it is due to differences in how eBook reading applications will process the content. For example here's how it looks on mine:

![ebook](https://user-images.githubusercontent.com/762512/118392321-64796300-b639-11eb-91b4-8cb566e74f80.png)

For me the font sizes are fine and there are line breaks appropriately placed... :crying_cat_face: 

I'll see if I can find a way to make it look better across multiple apps.

**Comment by MugUK** [comment]
I just picked and installed the ePub reader that a site recommended. Guess I picked a crap one at the same time :(

Maybe add a caveat somewhere that "not all ePub readers will display the pages in the same way *cough* SumatraPDF we're looking at you *cough* " :)

Is it as easy to add an export to PDF or does the code you're using not include that?

**Comment by MugUK** [comment]
Just spotted that your ereader and this Chrome pluging I'm trying out both say "Scolltexts of x y z"?

**Comment by MugUK** [comment]
As nice as this ePub reader extension for Chrome makes the books display neatly ... you can't save off the entire ebook as a PDF. It's limited to 1 page :(

[ePub Reader for Google Chrome™ __ Scolltexts of Awesome_Elite (en).pdf](https://github.com/atari-legend/atari-legend/files/6489059/ePub.Reader.for.Google.Chrome.__.Scolltexts.of.Awesome_Elite.en.pdf)

**Comment by nguillaumin** [comment]
I did a bit more research, basically there are 2 types of reader apps I think: The ones that use a web-browser engine and will respect the styles properly, and the ones that do not use one and have to re-implement the styling themselves. For the latter usually the style is all messed up as it does not support most of the CSS features. I found a similar one to yours on Linux, "FBReader" where the result was not very nice either...

I looked at trying to use less styling and put the spacing directly in the HTML (With `<br>`, `<center>` tags and such) but of course it breaks things on apps and devices that support the "proper" styling! Especially tags like `<center>` are not permitted anymore... So I think it will have to stay like that, I'd rather use proper styling and have it work on the devices (like actual ebook readers) than hack a way for the "dodgy" reader apps that don't implement the ePub spec properly.

On some specific issues you mentioned:
- "Continued on next page": Impossible unfortunately because the page flow is controlled by the reader itself, as it depends on the reader screen size. So I don't know where the page break is (actually it even depends on the font size, which can be controlled by the user). I don't think it's a problem it will be obvious to people used to using e-readers I think
- Missing menus: The `MISSING!` text you saw is actually the scrolltext :wink: Missing menus will be displayed with "[Unknown contents]" and no screenshots. See:

![1](https://user-images.githubusercontent.com/762512/118394641-220a5300-b646-11eb-8d5b-7330bd22e5dc.png)

- Bad page numbers: Looks to be a bug of the app, page numbers are correct as far as I can see on my other devices / apps.
- PDF:  Unfortunately this would be a completely different beast, and not very easy :disappointed:
